core/templates/domain/topic/SubtopicPageObjectFactory.ts

```diff
@@ -22,28 +22,27 @@ import cloneDeep from 'lodash/cloneDeep';
 import { Injectable } from '@angular/core';
 import { downgradeInjectable } from '@angular/upgrade/static';
 
-import { SubtopicPageContentsObjectFactory } from
-  'domain/topic/SubtopicPageContentsObjectFactory';
-
-export interface ISubtopicPage {
-  getId: () => string;
-  getTopicId: () => string;
-  getPageContents: () => void;
-  getLanguageCode: () => string;
+import {
+  ISubtopicPageContentsBackendDict,
+  SubtopicPageContents,
+  SubtopicPageContentsObjectFactory
+} from 'domain/topic/SubtopicPageContentsObjectFactory';
+
+interface ISubtopicPageBackendDict {
+  'id': string;
+  'topic_id': string;
+  'page_contents': ISubtopicPageContentsBackendDict;
+  'language_code': string;
 }
 
 export class SubtopicPage {
   _id: string;
   _topicId: string;
-  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
-  // 'any' because '_pageContents' is a dict with underscore_cased
-  // keys which give tslint errors against underscore_casing in favor of
-  // camelCasing.
-  _pageContents: any;
+  _pageContents: SubtopicPageContents;
   _languageCode: string;
   constructor(
-      subtopicPageId: string, topicId: string, pageContents: any,
-      languageCode: string) {
+      subtopicPageId: string, topicId: string,
+      pageContents: SubtopicPageContents, languageCode: string) {
```

Comment:
Please make these underscored-fields private, and use constructor w/ modifiers to simplify this definition:

```suggestion
  constructor(
      private id: string,
      private topicId: string,
      private pageContents: SubtopicPageContents,
      private languageCode: string) {}
```

Reply:
done